src/cc/bcc_syms.cc

```diff
@@ -0,0 +1,258 @@
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <unordered_map>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "bcc_helpers.h"
+
+class SymbolCache {
+public:
+	virtual void refresh() = 0;
+	virtual bool resolve_addr(uint64_t addr, struct bcc_symbol *sym) = 0;
+	virtual bool resolve_name(const char *name, uint64_t *addr) = 0;
+};
+
+class KSyms : SymbolCache {
+	struct Symbol {
+		Symbol(const char *name, uint64_t addr) :
+			name(name), addr(addr) {}
+		std::string name;
+		uint64_t addr;
+
+		bool operator<(const Symbol& rhs) const { return addr < rhs.addr; }
+	};
+
+	std::vector<Symbol> _syms;
```

Comment:
I'm more used to the google C++ style guide, which is a bit old but easy to verify (single python script). I am considering adding their cpplint.py to our tree and wrapping it in a ctest.